### PR TITLE
Feature/minvp protect routes

### LINKED ISSUE #185: Refactor web app to separate MinVP from MaxVP based on release flags

### PR DESCRIPTION
**Pull request by ad2969**
🎟️ Ticket(s): Closes #185 

👷 Changes: A brief summary of what changes were introduced.
1. Introduces `REACT_APP_SHOW_MAXVP` flag into the app to hide routes and buttons that we don't want to see in the current MinVP (basically everything front-facing)
    --> mainly contains changes in `routes.js` and `nav.js`
2. Fixing travis build (`node-sass` doesn't seem to be supported in Node v15 builds yet...) The PR reverts node to v12 (the most recent stable release)

## Checklist (PR does not include UI/styling changes)
- [x] Looks good on large screens
- [x] Looks good on mobile

### CODE REVIEW

File: src/components/Nav.js [hunk 1]
```diff
@@ -55,63 +64,65 @@ function Nav (props) {
...
+          onClick={handleItemClick.bind(null, '/events')}
+          bar={selectedItem === '/events' ? barSelected : barUnselected}
+          featureFlag='REACT_APP_SHOW_MAXVP'
+        />
+        <MenuItem
+          label='Profile'
+          icon={<PersonIcon style={selectedItem === '/profile' ? selected : unselected} />}
+          onClick={handleItemClick.bind(null, '/profile')}
+          bar={selectedItem === '/profile' ? barSelected : barUnselected}
+          featureFlag='REACT_APP_SHOW_MAXVP'
```
Comment (dchen150) [inline src/components/Nav.js:118]: is there any reason to use the feature flag on each individual item on the nav bar? would it not be sufficient to do something like: 
`isFeatureEnabled ? // show member nav bar : // do not show member nav bar`

File: src/components/Router.js [hunk 2]
```diff
@@ -105,88 +105,93 @@ class Router extends Component {
...
+            component={EventRegister} />
+          <Route
+            path='/login-redirect'
+            component={LoginRedirect} />
+          <Route
+            path='/signup'
+            featureFlag={'REACT_APP_SHOW_MAXVP'}
+            component={Signup} />
+          <Route
+            path='/'
```
Comment (dchen150) [inline src/components/Router.js:188]: should this be set to `/exec`? or maybe could be in a different PR

File: src/components/Nav.js (hunk 1 above)
Comment (ad2969) [reply src/components/Nav.js:118]: There's not a huge reason why... I just felt it was cleaner than having too many ternary operators in the render function...

File: src/components/Router.js (hunk 2 above)
Comment (ad2969) [reply src/components/Router.js:188]: This is a good point..
In the [refactor](https://github.com/ubc-biztech/bt-web/pull/187) I tried to address this issue by standardizing the namespace "exec" to "admin", but this route was changed to simply `login`.

Also I realize I think I missed something here. One addition that we should probably have is to check the `REACT_APP_SHOW_MAXVP` feature flag and disallow login if the user is not a `@ubcbiztech.com`?

Comment (dchen150) [reply src/components/Router.js:188]: oh yes, good catch LOL, should not let anyone sign in unless they are a biztech exec

**Review by jacqueschen1 — DISMISSED** [review]
Not too familiar with this code (since mostly worked on backend), but looks good. Nit: could you add in the description/title that there was also a bunch of linting done?

**Review by dchen150 — APPROVED** [review]
